### PR TITLE
Add sense of session using the header component

### PR DESCRIPTION
**Pull request by duranmla**
**What:**
Update the header and pages on the site so they reflect if a user is logged in

**Why:**
This is related to the milestone at https://github.com/debtcollective/student-debt-campaign/issues/3.
Closes https://github.com/debtcollective/student-debt-campaign/issues/23
Closes https://github.com/debtcollective/student-debt-campaign/issues/24

**How:**
Check https://github.com/debtcollective/student-debt-campaign-netlify/commit/8a0c9b3cf4fc7fd499c963da3cf10d727e552f3d

**Notes:**
Besides, the main code to have the sense of sessions, we have added a lot of extra work on setup for Apollo client, prettier, jest and aside setup that needed to be done and that somehow affected the development of this feature.

1. The current problem we have is to being able to trigger the logout from the site itself _(that's the only thing missing for this milestone)_ Discourse doesn't support a link to trigger the logout, maybe we can add the endpoint using the current SSO plugin?
2. We need to work on private routes in a separate PR in order to redirect users from private pages to home if they are not logged in

## Screenshots
https://www.loom.com/share/d4f673597e5a40d1ad99614f68f3c9ec

### CODE REVIEW

File: src/components/Footer/index.js [hunk 1]
```diff
@@ -109,7 +108,7 @@ const Footer = () => {
             <a href="/">
               <img
                 alt="The Debt Collective"
-                src={logo}
+                src="/img/logo-dark.svg"
```
Comment (orlando) [inline src/components/Footer/index.js:111]: @duranmla why was this changed?

Comment (duranmla) [reply src/components/Footer/index.js:111]: https://github.com/debtcollective/student-debt-campaign-netlify/pull/23/commits/9b97d3cbc07fda3845dedc3eedc142202d6faa83 basically I have an issue with the file-mock that didn't allow me to run my tests on jest using the `--watch` mode.

This file https://github.com/gatsbyjs/gatsby/blob/master/examples/using-jest/__mocks__/file-mock.js is created to say to jest here https://github.com/gatsbyjs/gatsby/blob/master/examples/using-jest/jest.config.js#L7 that we need to "ignore" some assets such as svg BUT placing those files outside the static folder  make this to not work even when we used to have https://github.com/debtcollective/student-debt-campaign-netlify/pull/23/commits/927ea4fce4236eb078de07a49a0d785d8543a126

**TL;DR** it was a workaround to avoid to have the issue I had with jest, since the configuration was done but not working for a reason that I couldn't find.

File: src/components/Header/index.js [hunk 2]
```diff
@@ -82,39 +116,45 @@ const Header = () => {
...
-                >
-                  Login
-                </a>
+                {isLoggedIn ? (
+                  <Profile user={user} />
+                ) : (
+                  <>
+                    {/* >= lg */}
+                    <a
+                      href={`${process.env.GATSBY_COMMUNITY_URL}/session/sso_cookies/signup?${redirectParam}`}
```
Comment (orlando) [inline src/components/Header/index.js]: @duranmla I suggest two variables here instead of hard-coding the URL, one for login and one for signup. These URLs are not stable yet and may change.

File: src/lambda/hello.js [hunk 3]
```diff
@@ -1,13 +1,13 @@
 // For more info, check https://www.netlify.com/docs/functions/#javascript-lambda-functions
 export function handler(event, context, callback) {
-  console.log('queryStringParameters', event.queryStringParameters)
+  console.log("queryStringParameters", event.queryStringParameters);
```
Comment (orlando) [inline src/lambda/hello.js]: Let's remove this file

File: src/templates/actions-page.js [hunk 4]
```diff
@@ -1,20 +1,35 @@
...
 import Campaign from "../sections/CampaignActions";
+import { GET_USER } from "../api";
 
-// TODO: this data needs to be pulled from our SSO service
-const tempFakeUser = {
-  id: "3",
-  name: "Jane Doe",
-  email: "jane.doe@mail.com"
-};
+// TODO: this campaignId needs to be pull from the server (mean while take from userActions table vs currentUser id)
```
Comment (orlando) [inline src/templates/actions-page.js:9]: @duranmla Actually is fine if we hard code this as an env variable. 

We should add seeds in Knex to create the default campaign

File: src/templates/actions-page.js [hunk 5]
```diff
@@ -1,20 +1,35 @@
 import React from "react";
+import { useQuery } from "@apollo/react-hooks";
 
 import Layout from "../components/Layout";
+import Header from "../components/Header";
 import Campaign from "../sections/CampaignActions";
```
Comment (orlando) [inline src/templates/actions-page.js]: @duranmla should we name this `CampaignActions` instead?

**Review by orlando — CHANGES_REQUESTED** [review]
some small comments

**Comment by orlando** [comment]
@duranmla about your comments

For **logout** can be done by sending a XHR destroy request to Discourse `/logout` route and it should work. This is how we are currently doing for the tools app https://github.com/debtcollective/dc-vue-header/blob/341ff1efb8778296682b10a1fef2324bd8e4f90f/src/services/ProfileService.js#L32-L40

File: src/templates/actions-page.js (hunk 4 above)
Comment (duranmla) [reply src/templates/actions-page.js:9]: Let's create an issue within this project to address this one.
nevertheless, with the `UserCampaign` we can pull the campaign from there too

Comment (duranmla) [reply src/templates/actions-page.js:9]: I have clarified this already. with @orlando helps
There are few things to understand:
1. The responsibility about knowing which is the currentCampaign is from the server _(so the hard code takes place there)_
2. We already know about which is the campaign, in that sense, we can have a resolver that look for the _first campaign_ and return it.
3. Client probably doesn't need to request the currentCampaign because the requests are going to be scoped already by the server

**Comment by duranmla** [comment]
@orlando updates has taken place already. Let me know.

**Review by orlando — APPROVED** [review]
LGTM

File: src/templates/actions-page.js (hunk 5 above)
Comment (orlando) [reply src/templates/actions-page.js]: @duranmla I think you missed this change

**Comment by orlando** [comment]
Merging this to change the repo name
